src/drive/components/AppRoute.jsx

```diff
@@ -16,7 +17,12 @@ const AppRoute = (
     <Redirect from="/files/:folderId" to="/folder/:folderId" />
     <Route component={FileExplorer}>
       <Redirect from="/" to="folder" />
-      <Route path="folder(/:folderId)" component={Folder} />
+      <Route path="folder" component={Folder}>
+        <Route path="file/:fileId" component={FileViewer} />
+      </Route>
+      <Route path="folder/:folderId" component={Folder}>
```

Comment:
I think we avoid the duplication here by using an optional parameter, such as `<Route path="folder/(:folderId)" component={Folder}>`

Reply:
I tried, but it doesn't work: apparently an optional parameter must be on a leaf route and can't have children